modules/chapter4/pages/section4.adoc

```diff
@@ -1,6 +1,191 @@
+
+. Create the object bucket claim using the `openshift-storage.noobaa.io` storage class.
++
+[source,bash,role=execute]
+----
+cat <<EOF | oc apply -f -
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: noobaatest
+  namespace: openshift-storage
+spec:
+  storageClassName: openshift-storage.noobaa.io
+  generateBucketName: noobaatest
+EOF
+----
+
+. Get the bucket name by entering the following command.
++
+[source,bash,role=execute]
+----
+bucket_name=$(oc get obc -n openshift-storage noobaatest -o jsonpath='{.spec.bucketName}')
+----
+
+. Get the AWS credentials by entering the following commands.
++
+[source,bash,role=execute]
+----
+AWS_ACCESS_KEY_ID=$(oc get secret -n openshift-storage noobaatest -o yaml | grep -w "AWS_ACCESS_KEY_ID:" | head -n1 | awk '{print $2}' | base64 --decode)
+----
++
+[source,bash,role=execute]
+----
+AWS_SECRET_ACCESS_KEY=$(oc get secret -n openshift-storage noobaatest -o yaml | grep -w "AWS_SECRET_ACCESS_KEY:" | head -n1 | awk '{print $2}' | base64 --decode)
+----
+
+. Create the secret `image-registry-private-configuration-user` with the AWS credentials for the new bucket under `openshift-image-registry` project by entering the following command.
++
+[source,bash,role=execute]
+----
+oc create secret generic image-registry-private-configuration-user --from-literal=REGISTRY_STORAGE_S3_ACCESSKEY=${AWS_ACCESS_KEY_ID} --from-literal=REGISTRY_STORAGE_S3_SECRETKEY=${AWS_SECRET_ACCESS_KEY} --namespace openshift-image-registry
+----
+
+. Get the route host by entering the following command.
++
+[source,bash,role=execute]
+----
+route_host=$(oc get route s3 -n openshift-storage -o=jsonpath='{.spec.host}')
+----
+
+. Create a config map that uses an ingress certificate by entering the following commands.
++
+[source,bash,role=execute]
+----
+oc extract secret/router-certs-default -n openshift-ingress --confirm
+----
++
+[source,bash,role=execute]
+----
+oc create configmap image-registry-s3-bundle --from-file=ca-bundle.crt=./tls.crt  -n openshift-config
+----
+
+. Configure the image registry to use the `Nooba` object storage by entering the following command:
++
+[source,bash,role=execute]
+----
+oc patch config.image/cluster -p '{"spec":{"managementState":"Managed","replicas":2,"storage":{"managementState":"Unmanaged","s3":{"bucket":'\"${bucket_name}\"',"region":"us-east-1","regionEndpoint":'\"https://${route_host}\"',"virtualHostedStyle":false,"encrypt":false,"trustedCA":{"name":"image-registry-s3-bundle"}}}}}' --type=merge
+----
+
+. Notice the `image-registry` cluster operator is in progressing state.
++
+.Sample output:
+----
+[root@hypervisor ~]# oc get co image-registry
+NAME             VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
+image-registry   4.16.8    False       True          False      13s     Available: The deployment does not have available replicas...
+----
+
+. In a minute, `image-registry` cluster operator is available.
```

Comment:
"In a minute, 'image-registry' cluster operator will be available."

Reply:
Done.